_posts/2018/2018-05-27-first-and-last-may-not-mean-what-you-think.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Active Record’s first and last may not mean what you think
+description: 'Naming truly is an art'
+layout: article
+category: ruby
+image:
+  base: '2018/first-and-last-may-not-mean-what-you-think'
+  alt: 'Running track'
+  source: 'https://unsplash.com/photos/52p1K0d0euM'
+  credit: 'Austris Augusts'
+---
+
+Each ActiveRecord model comes with the well used ‘get me a single record’ scopes; `.first` & `.last`.
+
+## Instead of…
+
+...using the methods directly on the base ActiveRecord class.
+
+```ruby
+User.first
+User.last
+```
+
+
+## Use…
+
+...them only with ordered scopes. Ideally with named concepts.
+
+### `app/models/user.rb`
+
+```ruby
+class User < ActiveRecord::Base
+  scope :by_created, -> { order(created_at: :asc) }
+  scope :earliest_created, -> { by_created.first }
+  scope :most_recently_created, -> { by_created.last }
+  # ...
+end
+
+User.most_recently_created
+```
+
+
+## But why?
+
+I’ll admit, this is a subtle one. My argument is mostly about making the code more explicit.
+
+If you’re using the bare `.first` and `.last` methods on an bare ActiveRecord model scope you are relying heavily on the Rails convention of an incrementing integer ID.
+
+Those methods are thus colloquially used to mean ‘created first’ and ‘created most recently’. It's only an accident of Rails default design that sorting by ascending `id` is same order as the ascending `created_at` timestamp.
```

Comment:
I get what you are saying here but I think you can make this slightly more straightforward.

I think there's some remixing involved between this paragraph and the one above. You want to first talk about what people think the '.first' and '.last' methods mean — and then go into how that falsely relies on the incrementing integer and the fact that it's the same ascending order. Does that make sense?

It seems as if there's a story in these two paragraphs/a thought process, that's slightly jumbled up.